### PR TITLE
Add the Variable struct

### PR DESCRIPTION
**Pull request by Nepomuceno**
Adding a struct to represent the variable block in the terraform files
This will help in storing this values while parsing tf files.

### CODE REVIEW

File: module/variable.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package module
+
+import (
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type Variable struct {
```
Comment (radeksimko) [inline module/variable.go:9]: Do you mind adding `Variables` map leveraging this to `module.Meta`?

File: module/variable.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	Description     string
+	Type            cty.Type
+	TypeConstraints schema.ExprConstraints
```
Comment (radeksimko) [inline module/variable.go:12]: What do we need `TypeConstraints` here for? 🤔

File: module/variable.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	Default         hcl.Expression
```
Comment (radeksimko) [inline module/variable.go:13]: I reckon we'll need this eventually for https://github.com/hashicorp/terraform-ls/issues/495 - good call adding it, but I think we could already store it as `cty.Value` and call the field `DefaultValue`?

File: module/variable.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+	// In case the version it is before 0.14 sensitive will always be false
+	// that was actually the default value for prior versions
+	Sensitive bool
```
Comment (radeksimko) [inline module/variable.go:17]: Nitpick: Do you mind calling it `IsSensitive`?

**Review by radeksimko — COMMENTED** [review]
This is a great start! 👍🏻 

Aside from my in-line comment - I think that we can also leverage the new struct in the `earlydecoder` and add a test for the collection of variables there?
https://github.com/hashicorp/terraform-schema/blob/main/earlydecoder/decoder.go

Then you could raise a separate PR to `terraform-ls` which implements steps 3 and 4 of https://github.com/hashicorp/terraform-ls/issues/50

and then finally carry on with the rest of https://github.com/hashicorp/terraform-ls/issues/537

File: module/variable.go (hunk 3 above)
Comment (Nepomuceno) [reply module/variable.go:13]: Didn't know cty.Value did exist 😁changing it to cty.Value

File: module/variable.go (hunk 2 above)
Comment (Nepomuceno) [reply module/variable.go:12]: ExprConstraints will be needed when you specify the schema you can derive them from the cty.Type but making this conversion every time we need that could be expensive that is why I thought about being able to store this in the variables.

File: module/variable.go (hunk 1 above)
Comment (Nepomuceno) [reply module/variable.go:9]: Got it makes more sense to put that there them instead of where I was adding that directly into terraform-ls

Comment (radeksimko) [reply module/variable.go:8]: We'll need them in both places actually - which may look like a duplication, but this helps us avoid any unnecessary dependencies between the two.

File: module/variable.go (hunk 2 above)
Comment (radeksimko) [reply module/variable.go:11]: I think that creating one slice with a single element is negligible when we have to modify the schema anyway? I don't mind adding it if we conclude that it makes a _significant_ impact on the UX/CPU/memory, but I'd prefer to start with a simpler and cleaner API and work from there, rather than "optimize early", if possible.

Comment (Nepomuceno) [reply module/variable.go:11]: Perfect will remove and we can optimize it later.

**Review by radeksimko — APPROVED** [review]

**Comment by radeksimko** [comment]
I'm going to have #48 reviewed first and then once merged I'd like to revisit this PR, rebase it and see if we can introduce the `typeexpr` fork here and use it to parse variables in the appropriate format into these new structs.

**Review by radeksimko — COMMENTED** [review]
LGTM, aside from my in-line comments (most of which are nitpicks) 😉

File: earlydecoder/load_module.go [hunk 5]
```diff
@@ -166,7 +170,35 @@ func loadModuleFromFile(file *hcl.File, mod *decodedModule) hcl.Diagnostics {
 					LocalName: inferProviderNameFromType(r.Type),
 				}
 			}
+
+		case "variable":
+			content, _, contentDiags := block.Body.PartialContent(variableSchema)
+			diags = append(diags, contentDiags...)
+			name := block.Labels[0]
```
[the remark this answers is not in the document]
Comment (radeksimko) [reply earlydecoder/load_module.go:180]: Also can you add a test for these edge cases, please? i.e. a variable block with non-empty 2+ labels, empty label (`variable "" { ... }`) and one with label-less `variable` block (`variable {}`).

**Review by radeksimko — APPROVED** [review]
